cms/djangoapps/contentstore/views/library.py

```diff
@@ -50,6 +50,11 @@ def library_handler(request, library_key_string=None):
         log.exception("Attempted to use the content library API when the libraries feature is disabled.")
         raise Http404  # Should never happen because we test the feature in urls.py also
 
+    if not CourseCreatorRole().has_user(request.user):
```

Comment:
You should just call `_get_library_creator_status` here instead of duplicating the logic. What's here is actually different, which will cause bugs. For example, this code does not check if `ENABLE_CREATOR_GROUP` is active or not.